packages/rum-core/src/domain/rumEventsCollection/action/trackActions.ts

```diff
@@ -65,142 +54,176 @@ export interface ActionContexts {
 export const AUTO_ACTION_MAX_DURATION = 10 * ONE_SECOND
 export const ACTION_CONTEXT_TIME_OUT_DELAY = 5 * ONE_MINUTE // arbitrary
 
-interface ActionController {
-  id: string
-  startClocks: ClocksState
-  discard(): void
+interface TrackActionsState {
+  readonly lifeCycle: LifeCycle
+  readonly domMutationObservable: Observable<void>
+  readonly actionNameAttribute: string | undefined
+  readonly history: ContextHistory<string>
+  readonly stopObservable: Observable<void>
 }
 
 export function trackActions(
   lifeCycle: LifeCycle,
   domMutationObservable: Observable<void>,
   { actionNameAttribute }: RumConfiguration
 ) {
-  const history = new ContextHistory<ActionController>(ACTION_CONTEXT_TIME_OUT_DELAY)
+  const state: TrackActionsState = {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I am not quite confortable with the "state" naming. I would associate that to something that we ave computed internally and can change, like what we have for `SessionState` or `RecorderState`. 
FMU here, it is more a bundle of external and internal dependencies.
What was your intent?

Reply:
I went with closures, let me know what you think